File: cpp/euler001/euler001.cpp
/**
 * @file    euler001.cpp
 * @author  Marvin Smith
 * @date    4/25/2015
 */
#include <iostream>

/**
 * @brief Main Function
 */
int main( int argc, char* argv[] ){

    // Define our running sum
    int sum = 0;

    // Iterate from 3 to 999
    for( int i=3; i<=999; i++ ){
        
        // Check if divisible by 3 or 5
        if( i%3 == 0 || i%5 == 0 ){
            sum += i;
        }
    }

    // Print Result
    std::cout << sum << std::endl;
    
    // Exit 
    return 0;
}

File: rust/src/euler001/euler001.rs

/// Main Driver
fn main() {

    let mut sum = 0;
    for x in 1..1000 {
        // Check if divisible by 3
        if ( x % 3 ) == 0 || 
           ( x % 5 ) == 0 {
            sum += x;
        }
    }
    println!( "Sum: {}", sum );
}